### ISSUE TITLE
Can't set headers after they are sent.

### ISSUE THREAD

**Issue by xgibbousx** [prerender/prerender#312]
I am running my own Prerender server locally along with the prerender-node library.  Everything is working perfectly on an initial request as seen by the Prerender server logs:

> server-1 (out): 2016-05-13T11:43:43.738Z getting http://statpedia.com/
> server-1 (out): 2016-05-13T11:43:45.614Z got 200 in 1876ms for http://statpedia.com/
> 
> server-1 (out): 2016-05-13T11:43:55.271Z getting http://statpedia.com/
> server-1 (out): 2016-05-13T11:43:55.910Z got 304 in 639ms for http://statpedia.com/

However, on subsequent requests, I get the following error message in my Prerender server logs:

> server-1 (out): 2016-05-13T11:43:55.917Z error setting header: [Error: Can't set headers after they are sent.]

No html is returned upon receiving the error above.  If I restart the Prerender server, the issue is temporarily resolved for just one or two requests then it goes back to the header errors.

Any ideas on how I can fix this problem?

Just got the following for the first time as well:

> server-1 Error: Can't set headers after they are sent.
> server-1     at ServerResponse.OutgoingMessage.setHeader (_http_outgoing.js:335:11)
> server-1     at Object.server._sendResponse (/var/www/statpedia.com/prerender/lib/server.js:522:17)
> server-1     at /var/www/statpedia.com/prerender/lib/server.js:509:37
> server-1     at Gzip.onEnd (zlib.js:227:5)
> server-1     at Gzip.emit (events.js:129:20)
> server-1     at _stream_readable.js:908:16
> server-1     at process._tickDomainCallback (node.js:381:11)

**Comment by xgibbousx** [comment]
Just got the following for the first time as well:

> server-1 Error: Can't set headers after they are sent.
> server-1     at ServerResponse.OutgoingMessage.setHeader (_http_outgoing.js:335:11)
> server-1     at Object.server._sendResponse (/var/www/statpedia.com/prerender/lib/server.js:522:17)
> server-1     at /var/www/statpedia.com/prerender/lib/server.js:509:37
> server-1     at Gzip.onEnd (zlib.js:227:5)
> server-1     at Gzip.emit (events.js:129:20)
> server-1     at _stream_readable.js:908:16
> server-1     at process._tickDomainCallback (node.js:381:11)

**Comment by thoop** [comment]
Sorry for the delay! Looks like some code path is being followed twice or something and setting the headers twice.

I'm not seeing that issue when I try locally:

```
2016-06-01T22:27:18.208Z getting http://statpedia.com/
2016-06-01T22:27:22.555Z got 200 in 4347ms for http://statpedia.com/
2016-06-01T22:27:25.858Z getting http://statpedia.com/
2016-06-01T22:27:26.177Z got 304 in 318ms for http://statpedia.com/
2016-06-01T22:27:45.658Z getting http://statpedia.com/
2016-06-01T22:27:45.975Z got 304 in 317ms for http://statpedia.com/
2016-06-01T22:27:50.787Z getting http://statpedia.com/
2016-06-01T22:27:51.410Z got 304 in 623ms for http://statpedia.com/
2016-06-01T22:27:56.146Z getting http://statpedia.com/
2016-06-01T22:27:56.463Z got 304 in 317ms for http://statpedia.com/
```

Which version of node are you using? Do you have any custom prerender plugins being used?

**Comment by thoop** [comment]
This should be fixed in Prerender v4.1.0. Please reopen if you continue to see issues.
